### PR TITLE
Fix Nix publishing

### PR DESCRIPTION
**Pull request by sagikazarmark**
This is attempt numero due to fix Nix package publishing to dagger/nix.

https://github.com/dagger/dagger/pull/6063 added Nix to the GitHub Actions workflow, but GoReleaser actually runs in Dagger using a GoReleaser Pro image.

This PR removes Nix from the GHA workflow and adds Nix to the GoReleaser image.

(Reminder: we only need `nix-prefetch-url` for GoReleaser to work, but it's not bundled with their image and installing Nix from the Alpine package repo seems to be the easiest way to get `nix-prefetch-url`)

### CODE REVIEW

**Review by gerhard — COMMENTED** [review]
Great catch @sagikazarmark 🙌

We are just about to run another release, fingers crossed that this fixed it 🤞

**Review by gerhard — APPROVED** [review]
Will merge when the last check passes 👍 🚀

**Comment by jedevc** [comment]
@gerhard checks passed, merging now :tada:

**Comment by jedevc** [comment]
Uh oh.

This fails on main: https://github.com/dagger/dagger/actions/runs/7288062871/job/19859939157. The step here wasn't run in this PR, so wasn't caught before. It would be nice if we could dry-run the publish job in PRs to avoid this.

It looks like `nix` is only available on `alpine:edge`: https://pkgs.alpinelinux.org/packages?name=nix&branch=edge&repo=&arch=&maintainer=, which go releaser doesn't use.

Currently looking at alternative installation methods for nix.

**Pull request merged**
